R/mutate_mzroll_list.R

```diff
@@ -4,55 +4,96 @@
 #'   undetected peaks.
 #'
 #' @inheritParams test_mzroll_list
-#' @param log2_floor_value minimum value to set for low abundance or
-#'   missing peaks
+#' @param floor_value minimum value to set for low abundance or missing peaks
 #' @param floor_var measurement variable to floor to \code{log2_floor_value}.
+#' @param mzrolldb_file_path (optional) path to mzrollDB file, only used for QQQ datasets.
+#' @param data_type parameter to differentiate data types.
 #'
 #' @return \code{\link{process_mzroll}}
 #'
 #' @examples
 #' floored_peaks <- floor_peaks(nplug_mzroll_augmented, 12)
+#' floored_peaks <- floor_peaks(mzroll_list, 100, "log2_abundance", "QQQ")
+
 #' @export
 floor_peaks <- function(mzroll_list,
-                        log2_floor_value = 12,
-                        floor_var = "log2_abundance") {
-  test_mzroll_list(mzroll_list)
-
+                         floor_value = 12,
+                         floor_var = "log2_abundance",
+                         mzrolldb_file_path = NULL,
+                         data_type = "QE") {
+  test_mzroll_list_local(mzroll_list)
+  
   valid_floor_var <- setdiff(
     mzroll_list$design$measurements$variable,
     c(mzroll_list$design$feature_pk, mzroll_list$design$sample_pk)
   )
-
+  
   checkmate::assertChoice(floor_var, valid_floor_var)
   checkmate::assertNumeric(mzroll_list$measurements[[floor_var]])
-  checkmate::assertNumber(log2_floor_value)
-
-  # summarize peaks associated with each peakgroup
-
-  missing_peaks <- tidyr::expand_grid(
-    groupId = mzroll_list$features$groupId,
-    sampleId = mzroll_list$samples$sampleId
-  ) %>%
-    dplyr::anti_join(
+  checkmate::assertNumber(floor_value)
+  
+  if (data_type == "QQQ") {
+    ## find missing peaks
```

Comment:
I think that this code does not need to be explicitly limited to `data_type == "QQQ"`, I think instead it should be limited to cases where there is a value in the `peakBackground` field that is `>0`.  Currently, the only time this is ever true is when `data_type == "QQQ"`. This might also end up being assessed on a per-compound basis, as some compounds might end up with a nonzero `peakBackground` value while others have a value of 0, in which case the imputation approach might not work correctly.

Reply:
Thanks for the suggestion @PMSeitzer does the ```peakBackground``` exist for non-QQQ datasets in the mzrollDB?